standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -1861,6 +1861,18 @@ struct ScheduledQueryProgressInfo{
   4: optional string errorMessage,
 }
 
+struct GetFileListRequest {
+  1: optional string catName,
+  2: optional string dbName,
+  3: optional string tableName,
+  4: optional list<string> partVals,
+  6: optional string validWriteIdList
+}
+
+struct GetFileListResponse {
+  1: required list<binary> fileListData
```

Comment:
Should be optional, so if we decide to change something then we can change easily.

Maybe based on @vihangk1 's suggestion, we should add a version number here for the FB data. So if we change content of the binary later it could be done easily

Reply:
Made it optional and added version number